src/Integration/ProjectToLanguageMapper.cs

```diff
@@ -46,37 +71,43 @@ public static class ProjectToLanguageMapper
         /// New code should call <see cref="GetAllBindingLanguagesForProject(EnvDTE.Project)"/> instead
         /// and handle the fact that there could be multiple supported languages.
         /// </returns>
-        [Obsolete("Use GetAllBindingLanguagesForProject instead")]
-        public static Language GetLanguageForProject(EnvDTE.Project dteProject)
+        private Language GetLanguageForProject(Project dteProject)
         {
             if (dteProject == null)
             {
                 throw new ArgumentNullException(nameof(dteProject));
             }
 
-            Guid projectKind;
-            if (!Guid.TryParse(dteProject.Kind, out projectKind))
+            if (!Guid.TryParse(dteProject.Kind, out var projectKind))
             {
                 return Language.Unknown;
             }
 
-            Language language;
-            if (KnownProjectTypes.TryGetValue(projectKind, out language))
+            if (KnownProjectTypes.TryGetValue(projectKind, out var language))
             {
                 return language;
             }
+
+            if (projectKind == OpenAsFolderProject)
+            {
+                var cmakeProjectFile = absoluteFilePathLocator.Locate("CMakeLists.txt");
+
+                if (!string.IsNullOrEmpty(cmakeProjectFile))
+                {
+                    return Language.Cpp;
+                }
+            }
+
             return Language.Unknown;
         }
 
         /// <summary>
         /// Returns all of the supported Sonar languages for the sepcified project or Unknown
```

Comment:
Miscellaneous: pre-existing typo "sepcified".

This doc comment now belongs on the interface method.